src/Bicep.Cli/Services/CompilationService.cs

```diff
@@ -131,6 +143,24 @@ static IEnumerable<IDiagnostic> GetModuleDiagnosticsPerFile(SourceFileGrouping g
                 .ToDictionary(bicepFile => bicepFile, bicepFile => GetModuleDiagnosticsPerFile(sourceFileGrouping, bicepFile, originalModulesToRestore));
         }
 
+        private static IReadOnlyDictionary<BicepFile, IEnumerable<IDiagnostic>> GetForceModulesRestoreDiagnosticsByBicepFile(SourceFileGrouping sourceFileGrouping, ImmutableHashSet<ModuleDeclarationSyntax> modulesToRestore)
```

Comment:
It seems `GetModuleRestoreDiagnosticsByBicepFile` and `GetForceModulesRestoreDiagnosticsByBicepFile` are almost identical, so it might be worth consolidating them (can pass `forceModulesRestore` as a parameter and eliminate the `if-else` in `RestoreAsync`).